pkg/controller/controller.go

```diff
@@ -742,6 +758,19 @@ func (p *csiProvisioner) prepareProvision(ctx context.Context, claim *v1.Persist
 		deletionAnnSecrets.namespace = provisionerSecretRef.Namespace
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeAttributesClass) && vacName != nil {
```

Comment:
vacName maybe an empty string

Reply:
We need to check existence of the VAC, not only if the vacName is an empty string or not.

Reply:
Added check against `*vacName == ""` in latest version